scanpy/tests/test_highly_variable_genes.py

```diff
@@ -136,14 +139,29 @@ def _check_pearson_hvg_columns(output_df: pd.DataFrame, n_top_genes: int):
     assert np.nanmax(output_df["highly_variable_rank"].to_numpy()) <= n_top_genes - 1
 
 
-def test_pearson_residuals_inputchecks(pbmc3k_parametrized_small):
-    adata = pbmc3k_parametrized_small()
+@pytest.mark.parametrize("array_type", ARRAY_TYPES)
+@pytest.mark.parametrize("dtype", ["float32", "int64"])
+def test_pearson_residuals_inputchecks(array_type, dtype):
+    # TODO: do we have a preferred way of making such a small dataset, wich the array types option?
```

Comment:
copied the `array_type(adata.X)` from other tests.
is this how we want to set up datasets to test through the combinations?

Reply:
I think so?